### ISSUE TITLE
modify proxy_get_header_map_value ABI to support returning values from multiple occurrences of the header

### ISSUE THREAD

**Issue by mpwarres** [proxy-wasm/spec#53]
The [proxy_get_header_map_value](https://github.com/envoyproxy/envoy/blob/788f26648d7ef73bd22b5ff6f343ac57fe923544/source/extensions/common/wasm/context.cc#L767-L768) hostcall currently only returns a single value, which doesn't accomodate headers that appear multiple times (such as [Set-Cookie](https://httpwg.org/specs/rfc6265.html#set-cookie).

The next Proxy-Wasm ABI minor version should update `proxy_get_header_map_value` or add a variant that supports returning multiple values.

**Comment by mpwarres** [comment]
@kyessenov FYI

**Comment by PiotrSikora** [comment]
Definitely! See: https://github.com/PiotrSikora/spec/tree/cc53262df056036427476b272fb8f2438aa7975f/abi-versions/vNEXT#proxy_get_map_values
